clients/rest-client-generator/src/main/java/uk/gov/justice/services/clients/rest/generator/strategy/ClientMediaTypeGeneration.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.services.clients.rest.generator.strategy;
+
+import static java.util.Collections.emptyList;
+
+import uk.gov.justice.raml.common.mapper.ActionMapping;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.raml.model.ActionType;
+import org.raml.model.MimeType;
+
+/**
+ * Media type generation strategy to support backwards compatibility.
+ */
+public class ClientMediaTypeGeneration implements ClientGenerationStrategy {
```

Comment:
I feel this is a better method, rather than littering the code with if statements for backward compatibility. This should be refactored and removed once contexts are converted.